### PR TITLE
Connection is_gray, disconnect if no address advert #1270

### LINKED ISSUE #1270: Return IP addresses in node overload

### PR DESCRIPTION
**Pull request by boostmultifruit**
Resolve #1270.

### CODE REVIEW

File: plugins/net_plugin/net_plugin.cpp [hunk 1]
```diff
@@ -2230,6 +2234,11 @@ namespace eosio {
          c->state = connection_state::connected;
       }
       if (msg.generation == 1) {
+         if (c->is_gray && c->protocol_version < proto_address_advertising) {
+            my_impl->close(c);
+			return;
```
Comment (afalaleev) [inline plugins/net_plugin/net_plugin.cpp]: Typo

File: plugins/net_plugin/net_plugin.cpp [hunk 2]
```diff
@@ -525,6 +525,7 @@ namespace eosio {
       handshake_message       last_handshake_sent;
       int16_t                 sent_handshake_count = 0;
       connection_state        state = connection_state::none;
+	  bool                    is_gray = false;
```
Comment (afalaleev) [inline plugins/net_plugin/net_plugin.cpp]: Typo

File: plugins/net_plugin/net_plugin.cpp [hunk 3]
```diff
@@ -1407,6 +1408,7 @@ namespace eosio {
    }
 
    void sync_manager::recv_handshake(const connection_ptr& c, const handshake_message& msg) {
+      c->state = connection_state::connected;
```
Comment (afalaleev) [inline plugins/net_plugin/net_plugin.cpp]: I don't understand this change. For what and why?

File: plugins/net_plugin/net_plugin.cpp [hunk 4]
```diff
@@ -2324,7 +2333,9 @@ namespace eosio {
 
       c->last_handshake_recv = msg;
       c->_logger_variant.reset();
-      sync_master->recv_handshake(c,msg);
+      if (!c->is_gray) {
```
Comment (afalaleev) [inline plugins/net_plugin/net_plugin.cpp]: I don't understand this change. For what and why?

Comment (boostmultifruit) [reply plugins/net_plugin/net_plugin.cpp]: @afalaleev It is to do not sync if node is gray.

**Pull request merged**